pkg/algorithm/hived_algorithm.go

```diff
@@ -252,56 +261,24 @@ func (h *HivedAlgorithm) DeleteAllocatedPod(pod *core.Pod) {
 	info := internal.ExtractPodBindInfo(pod)
-	klog.Infof("[%v]: deleting from node %v, GPUs %v", internal.Key(pod), info.Node, info.GpuIsolation)
+	klog.Infof("[%v]: Deleting allocated pod from affinity group %v...", internal.Key(pod), s.AffinityGroup.Name)
+	klog.Infof("[%v]: Deleting from node %v, GPUs %v", internal.Key(pod), info.Node, common.ToJson(info.GpuIsolation))
 
-	if group := h.allocatedAffinityGroups[s.AffinityGroup.Name]; group == nil {
-		klog.Errorf("[%v]: group %v not found when deleting pod", internal.Key(pod), s.AffinityGroup.Name)
+	if g := h.allocatedAffinityGroups[s.AffinityGroup.Name]; g == nil {
+		klog.Errorf("[%v]: Group %v not found when deleting pod", internal.Key(pod), s.AffinityGroup.Name)
 		return
 	} else {
-		var podIndex int32
-		for _, gms := range info.AffinityGroupBindInfo {
-			if gpuNumber := int32(len(gms.PodPlacements[0].PhysicalGpuIndices)); gpuNumber == s.GpuNumber {
-				podIndex = getPodIndex(gms.PodPlacements, info.Node, info.GpuIsolation[0])
-				if podIndex == -1 {
-					klog.Errorf("[%v]: pod placement not found in group %v: node %v, GPUs %v",
-						internal.Key(pod), s.AffinityGroup.Name, info.Node, info.GpuIsolation)
-					return
-				}
-			}
+		if podIndex := getAllocatedPodIndex(info, s.GpuNumber); podIndex == -1 {
+			klog.Errorf("[%v]: Pod placement not found in group %v: node %v, GPUs %v",
+				internal.Key(pod), s.AffinityGroup.Name, info.Node, info.GpuIsolation)
+			return
+		} else {
+			g.allocatedPods[s.GpuNumber][podIndex] = nil
 		}
-		group.allocatedPods[s.GpuNumber][podIndex] = nil
-		if !group.gangReleaseEnable {
-			klog.Infof("[%v]: gang release NOT enabled for group %v, releasing resources for this pod",
-				internal.Key(pod), s.AffinityGroup.Name)
-			for _, gpu := range group.physicalGpuPlacement[s.GpuNumber][podIndex] {
-				if gpu != nil {
-					h.confirmReleasedGpu(gpu.(*PhysicalCell), group)
-				}
-			}
-		}
-
-		if allPodsReleased(group.allocatedPods) {
-			if group.gangReleaseEnable {
-				klog.Infof("[%v]: gang release enabled for group %v, releasing resources for all the pods",
-					internal.Key(pod), s.AffinityGroup.Name)
-			}
-			// Even if the group does not enable gang release, we still need to check the placements of all pods
-			// and release them if still in use (this could be because a pod in the group is never added,
-			// and hence also never deleted, we need to release the resource explicitly)
-			for _, podPlacements := range group.physicalGpuPlacement {
-				for _, podPlacement := range podPlacements {
-					for _, gpu := range podPlacement {
-						if gpu != nil && gpu.GetPriority() != freePriority {
-							h.confirmReleasedGpu(gpu.(*PhysicalCell), group)
-						}
-					}
-				}
-			}
-			delete(h.allocatedAffinityGroups, s.AffinityGroup.Name)
-			klog.Infof("[%v]: All pods complete, affinity group deleted: %v", internal.Key(pod), s.AffinityGroup.Name)
+		if allPodsReleased(g.allocatedPods) {
+			h.deleteAllocatedAffinityGroup(g, pod)
 		}
```

Comment:
Did you remove the Partial Release feature in this PR?

Reply:
Yes